### PR TITLE
Scripts: Add WordPress folder to the list of ignores in configs

### PR DESCRIPTION
**Pull request by gziolo**
## Description
Follow-up for #17004 where `wp-scripts env install` adds `wordpress` folder with the local instance of WordPress.

This PR adds this new `wordpress` folder to all recommended configs in `@wordpress/scripts` to ensure that this folder is ignored and doesn't trigger any false alarms. This is quite similar to what happens in Gutenberg with the config overrides:
https://github.com/WordPress/gutenberg/blob/f198997e2c8e377423beb230ce5283914076d396/.eslintignore#L9
https://github.com/WordPress/gutenberg/blob/f198997e2c8e377423beb230ce5283914076d396/.stylelintignore#L1

## Testing

Basically all the following should work as before:

`npm run lint-css`
`npm run lint-js`
`npm run lint-pkg-json`
`npm run test-e2e`
`npm run test-unit`

I did some testing by adding files with failures to the `wordpress` folder.

### CODE REVIEW

File: .stylelintignore [hunk 1]
```diff
@@ -1 +0,0 @@
-wordpress
```
Comment (gziolo) [inline .stylelintignore]: It now uses the recommended Stylelint config located in `@wordpress/scripts` package.

File: packages/e2e-tests/jest.config.js [hunk 2]
```diff
@@ -10,6 +10,8 @@ module.exports = {
 		'expect-puppeteer',
 	],
 	testPathIgnorePatterns: [
+		'/node_modules/',
```
Comment (gziolo) [inline packages/e2e-tests/jest.config.js]: See:
https://jestjs.io/docs/en/configuration#testpathignorepatterns-array-string

which has default: `["/node_modules/"]`

File: .stylelintignore (hunk 1 above)
Comment (ntwb) [reply .stylelintignore]: The `wordpress` folder is **not** included in the `.stylelintignore` file:
> https://github.com/WordPress/gutenberg/blob/master/packages/scripts/config/.stylelintignore#L3

Comment (ntwb) [reply .stylelintignore]: Ha, yes it is, its added in this PR, ignore the above 😏

**Review by ntwb — APPROVED** [review]

File: .stylelintignore (hunk 1 above)
Comment (gziolo) [reply .stylelintignore]: The question is whether it shouldn't be:

`/wordpress` as otherwise, it might trigger at any level of nesting. I don't know really what's the best way to proceed.

Comment (ntwb) [reply .stylelintignore]: I've got an idea on resolving this for the `@wordpress/scripts` package so that the configs can be "nested" and "inherited" in a better way than currently occurs.

Happy to see this PR land and any issues arise can be fixed in subsequent PRs

**Review by pento — APPROVED** [review]
👍🏻

File: .stylelintignore (hunk 1 above)
Comment (gziolo) [reply .stylelintignore]: Cool, looking forward to see your proposal.
